assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * AdSense Setup UseSnippet Switch Component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffectOnce } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADSENSE } from '../../../datastore/constants';
+import { parseAccountID } from '../../../util/parsing';
+import { UseSnippetSwitch } from '../../common';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupUseSnippetSwitch() {
+	const originalUseSnippet = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getOriginalUseSnippet()
+	);
+	const existingTag = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getExistingTag()
+	);
+	const clientID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClientID()
+	);
+
+	const { setUseSnippet } = useDispatch( MODULES_ADSENSE );
+
+	useEffectOnce( () => {
+		if ( existingTag ) {
+			setUseSnippet( false );
+		}
+	} );
+
+	if (
+		( originalUseSnippet && ! existingTag ) ||
+		undefined === existingTag ||
+		undefined === originalUseSnippet
+	) {
+		return null;
+	}
+
+	let checkedMessage, uncheckedMessage;
+	if ( existingTag === clientID ) {
+		// Existing tag with permission.
+		checkedMessage = __(
+			'You’ve already got an AdSense code on your site for this account. We recommend you use Site Kit to place the code to get the most out of AdSense. Make sure to remove the existing AdSense code to avoid conflicts with the code placed by Site Kit.',
```

Comment:
@hussain-t Instead of duplicating the first part of the string, can we have it as a variable and then append the extra message to it?

Reply:
IMO, it's better to keep it separated since it is more explicit and clean. Even if we append the extra message, we will have to translate the string.

Reply:
Your comment makes sense, as you said, it's more clean and explicit. I'll let the merge reviewer take a look though :)